spec/compiler/loader/unix_spec.cr

```diff
@@ -28,10 +28,10 @@ describe Crystal::Loader do
     end
 
     it "parses file paths" do
-      expect_raises(Crystal::Loader::LoadError, /#{Dir.current}\/foobar\.o.+(No such file or directory|image not found)/) do
+      expect_raises(Crystal::Loader::LoadError, /#{Dir.current}\/foobar\.o.+(No such file or directory|image not found|Cannot open)/) do
         Crystal::Loader.parse(["foobar.o"], search_paths: [] of String)
       end
-      expect_raises(Crystal::Loader::LoadError, /#{Dir.current}\/foo\/bar\.o.+(No such file or directory|image not found)/) do
+      expect_raises(Crystal::Loader::LoadError, /(#{Dir.current}\/foo\/bar\.o|-lfoo\/bar\.o).+(No such file or directory|image not found|Cannot open)/) do
```

Comment:
This spec is about checking the system error messages to ensure the correct path was looked up.
The error message from glibc, musl and darwin have a form like `<path>: cannot open shared object file: No such file or directory`.
On FreeBSD it's `Cannot open "<path>"`.

In order to express the same intention, the regex should probably be something like this:
```suggestion
      expect_raises(Crystal::Loader::LoadError, /(#{Dir.current}\/foo\/bar\.o.+(No such file or directory|image not found)|Cannot open "#{Dir.current}\/foo\/bar\.o")/) do
```